### PR TITLE
Feat readable token

### PR DESCRIPTION
**Pull request by lucky-lusa**

### CODE REVIEW

File: iris-client-bff/src/main/java/iris/client_bff/config/TokenConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+public class TokenConfig {
+
+	@Value("${tokens.datLength: #{40}}")
+	private int datLength = 40;
+
+	@Value("${tokens.catLength: #{40}}")
+	private int catLength = 40;
+
+	/**
+	 * Reason for the default hear in code:
```
Comment (lucky-lusa) [inline iris-client-bff/src/main/java/iris/client_bff/config/TokenConfig.java:30]: *here ?

File: iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java [hunk 2]
```diff
@@ -56,13 +56,22 @@ public String announce() throws IRISAnnouncementException {
...
-		var oneWeekFromNow = Instant.now().plus(7, ChronoUnit.DAYS);
+		var oneWeekFromNow = Instant.now().plus(Period.ofWeeks(1));
+		return this.sendAnnouncement(domain, oneWeekFromNow);
+	}
+
+	@Override
+	public String announceExplicitToken(String connectionAuthorizationToken) throws IRISAnnouncementException {
+		var domain = connectionAuthorizationToken
+				+ "."
+				+ config.getPublicProxyDomain();
```
Comment (lucky-lusa) [inline iris-client-bff/src/main/java/iris/client_bff/proxy/EPSProxyServiceServiceClient.java:67]: This should be config.getTargetSubdomain() then. Apps will be able to recover the whole domain from CAT from token + subdomain dependent on abbreviation in token.

File: iris-client-bff/src/main/resources/application.properties [hunk 3]
```diff
@@ -51,12 +51,7 @@ eps-client.proxy-client-url=https://localhost:7766/jsonrpc
 proxy-service.eps-name=private-proxy-1
 proxy-service.target-subdomain=proxy.test-gesundheitsamt.de
 proxy-service.target-proxy=public-proxy-1
-
-# URLs of external services
-ext-app.dw.baseurl=https://digitales-wartezimmer.org/
-ext-app.dw.suburl-newcase=index-case
-
-hd.zip-code=70176
+proxy-service.public-proxy-domain=proxy.test.iris-gateway.de
```
Comment (lucky-lusa) [inline iris-client-bff/src/main/resources/application.properties:54]: Can be removed then

File: iris-client-bff/src/main/java/iris/client_bff/config/TokenConfig.java (hunk 1 above)
Comment (jekutzsche) [reply iris-client-bff/src/main/java/iris/client_bff/config/TokenConfig.java]: Ops, thank you very much, well done. :+1:

File: iris-client-bff/src/main/java/iris/client_bff/config/HealthDepartmentProperties.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+@EqualsAndHashCode(callSuper = false)
+public class HealthDepartmentProperties extends ValidatedProperties {
+
+	HealthDepartmentProperties(Validator validator) {
+		super(HealthDepartmentProperties.class, validator);
+	}
+
+	@NotBlank(message = "{missing.property.hd.abbreviation}") //
+	@Pattern(regexp = "^[0123456789ABCDEFabcdef]{3}$", message = "{missing.property.hd.abbreviation}")
+	private String abbreviation;
```
Comment (lucky-lusa) [inline iris-client-bff/src/main/java/iris/client_bff/config/HealthDepartmentProperties.java:24]: Should not be necessary when we pull the information from BS

**Review by lucky-lusa — COMMENTED** [review]
is good so far, the new changes are still to come

**Review by lucky-lusa — COMMENTED** [review]
I cant approve my own PR.

**Review by jekutzsche — APPROVED** [review]

**Pull request merged**